sync-template.sh

```diff
@@ -65,6 +71,29 @@ INHELP=""
 
 trap "rm -f ${STDOUT} ${STDERR}; cd ${HERE}" ABRT EXIT HUP INT QUIT
 
+if type -p hub > /dev/null 2>&1 ; then
+  if [[ -z "${USE_HUB:-}" ]]; then
+    USE_HUB=true
+  fi
+elif [[ "${USE_HUB:-false}" = true ]]; then
+  echo ERROR: there is no \"hub\" command line tool
+  echo see https://github.com/github/hub
+  echo e.g. on Fedora - dnf -y install hub
+  exit 1
```

Comment:
Small nitpick: It would be nicer for users to check all dependencies first and then fail with a list of all missing dependencies instead of checking each dependency individually. Then the user only needs to install the packages once instead of switching between running the script and install dependencies.